optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/InnerScoreDirector.java

```diff
@@ -77,7 +77,7 @@
      * @throws IllegalStateException if {@link #isConstraintMatchEnabled()} returns false
      * @see #getIndictmentMap()
      */
-    Map<String, ConstraintMatchTotal> getConstraintMatchTotalMap();
+    <Score_ extends Score<Score_>> Map<String, ConstraintMatchTotal<Score_>> getConstraintMatchTotalMap();
```

Comment:
IMO we really *should* add `<Score_>` on `InnerScoreDirector`. I could then start cleaning up *a lot more* of the internals.

Reply:
+1 as it would reduce the number of duplicate `Score_` type definitions on methods.

Reply:
I think it would be a good experiment, but it might cause more harm than good. In any case, only the experiment will (dis)prove it.